chia/full_node/block_store.py

```diff
@@ -294,6 +297,39 @@ async def get_full_blocks_at(self, heights: List[uint32]) -> List[FullBlock]:
                 ret.append(self.maybe_decompress(row[0]))
             return ret
 
+    async def get_generators_at(self, heights: List[uint32]) -> List[SerializedProgram]:
+        assert self.db_wrapper.db_version == 2
+
+        if len(heights) == 0:
+            return []
+
+        generators: Dict[uint32, SerializedProgram] = {}
+        heights_db = tuple(heights)
+        formatted_str = (
+            f"SELECT block, height from full_blocks "
+            f'WHERE in_main_chain=1 AND height in ({"?," * (len(heights_db) - 1)}?)'
+        )
+        async with self.db.execute(formatted_str, heights_db) as cursor:
+            async for row in cursor:
+                if self.db_wrapper.db_version == 2:
+                    block_bytes = zstd.decompress(row[0])
+                else:
+                    block_bytes = row[0]
+
+                try:
+                    gen = generator_from_block(block_bytes)
+                except BaseException:
+                    # this is defensive, on the off-chance that
+                    # generator_from_block() fails, fall back to the reliable
+                    # definition of parsing a block
+                    b = FullBlock.from_bytes(block_bytes)
```

Comment:
This is great.

Reply:
Maybe put a log message here or we might miss it